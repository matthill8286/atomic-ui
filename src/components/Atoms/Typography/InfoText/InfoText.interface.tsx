import React from 'react'
import { ThemeColors } from '@/types/theme'
import { TypoProps } from '../Typo/Typo.interface'

export type TypographyScaleInfo = 'small' | 'small-highlighted' | 'extra-small'

export interface InfoTextProps extends TypoProps, React.HTMLProps<HTMLElement> {
  scale?: TypographyScaleInfo
  color?: ThemeColors
}
