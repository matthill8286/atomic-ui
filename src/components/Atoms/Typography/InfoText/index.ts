export * from './InfoText'
export * from './InfoText.interface'
