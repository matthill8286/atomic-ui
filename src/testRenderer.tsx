import { mount, ReactWrapper } from 'enzyme'
import React from 'react'
import { MemoryRouter as Router } from 'react-router-dom'
import renderer from 'react-test-renderer'
import { ThemeProvider } from 'styled-components'
import { saiyanTheme } from '@/styles/sc-vars-saiyan'
import { Theme } from '@/types'

export function renderWithTheme<C extends React.Component, P = C['props'], S = C['state']>(
  tree: React.ReactElement<P>,
  theme = saiyanTheme
) {
  return renderer.create(<ThemeProvider theme={theme}>{tree}</ThemeProvider>).toJSON()
}

export function renderWithThemeAndRouter(tree: React.ReactElement, theme: Theme = saiyanTheme) {
  return renderer
    .create(
      <ThemeProvider theme={theme}>
        <Router>{tree}</Router>
      </ThemeProvider>
    )
    .toJSON()
}

export function renderWithRouter(tree: React.ReactElement) {
  return renderer.create(<Router>{tree}</Router>).toJSON()
}

export function mountWithTheme<C extends React.Component, P = C['props'], S = C['state']>(
  tree: React.ReactElement<P>
): ReactWrapper<P, S, C> {
  return mount(<ThemeProvider theme={saiyanTheme}>{tree}</ThemeProvider>)
}

export function mountWithThemeAndRouter<C extends React.Component, P = C['props'], S = C['state']>(
  tree: React.ReactElement<P>
): ReactWrapper<P, S, C> {
  return mount(
    <ThemeProvider theme={saiyanTheme}>
      <Router>{tree}</Router>
    </ThemeProvider>
  )
}
