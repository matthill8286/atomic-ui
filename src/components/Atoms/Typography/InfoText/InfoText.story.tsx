import { boolean, number, select, text } from '@storybook/addon-knobs'
import { storiesOf } from '@storybook/react'
import * as React from 'react'
import { saiyanTheme } from '@/styles/sc-vars-saiyan'
import { ThemeColors } from '@/types/theme'
import { InfoText } from '..'
import { TypographyScaleInfo } from './InfoText.interface'

storiesOf('Design System/Atoms/Typography', module).add('InfoText', () => {
  const sampleText = text('Example text', `Lorem ipsum dolor`)
  const scaleOptions = ['small', 'small-highlighted', 'extra-small']
  const scale = select('Type scale', scaleOptions, 'small') as TypographyScaleInfo
  const tagOptions: (keyof JSX.IntrinsicElements)[] = ['span', 'em', 'strong', 'div']
  const tagName = select('Tag name', tagOptions, 'span')
  const showColors = boolean('Enable Custom Color', false)
  const selectedColor = select('Color', Object.keys(saiyanTheme.color), 'black') as ThemeColors
  const limitLines = number('Limit Lines', 0)

  return (
    <InfoText
      tag={tagName}
      scale={scale}
      {...(showColors && { color: selectedColor })}
      limitLines={limitLines}>
      {sampleText}
    </InfoText>
  )
})
