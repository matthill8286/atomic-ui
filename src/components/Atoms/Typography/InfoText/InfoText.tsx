import React from 'react'
import { Typo } from '../Typo'
import { InfoTextProps } from './InfoText.interface'

export const InfoText: React.FC<InfoTextProps> = ({
  tag = 'span',
  scale = 'small',
  color = 'grey4',
  weight,
  fontSize,
  children,
  ...restProps
}) => {
  const isSmall = scale === 'small'
  const isExtraSmall = scale === 'extra-small'
  const isHighlighted = scale === 'small-highlighted'

  const presetFontSize = ((isSmall || isHighlighted) && 'xxs') || (isExtraSmall && 'xxxs')
  const presetFontWeight = ((isSmall || isExtraSmall) && 'regular') || (isHighlighted && 'semibold')

  return (
    <Typo
      tag={tag}
      color={color}
      fontSize={fontSize || presetFontSize || undefined}
      weight={weight || presetFontWeight || undefined}
      {...restProps}>
      {children}
    </Typo>
  )
}

InfoText.displayName = 'InfoText'
