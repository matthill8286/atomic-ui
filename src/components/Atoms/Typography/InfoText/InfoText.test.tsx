import React from 'react'
import { renderWithTheme } from '@/testRenderer'
import { InfoText } from './InfoText'

describe('InfoText', () => {
  it('renders correctly', () => {
    const tree = renderWithTheme(<InfoText scale="small" />)
    expect(tree).toMatchSnapshot()
  })
})
