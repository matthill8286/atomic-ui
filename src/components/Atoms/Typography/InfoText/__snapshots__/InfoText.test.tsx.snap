// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InfoText renders correctly 1`] = `
.c0 {
  color: #999999;
  font-size: 12px;
  line-height: 16px;
  font-family: "Oswald","Raleway",Arial,sans-serif;
  font-weight: 400;
  -webkit-letter-spacing: normal;
  -moz-letter-spacing: normal;
  -ms-letter-spacing: normal;
  letter-spacing: normal;
}

<span
  className="c0"
  color="grey4"
  fontSize="xxs"
/>
`;
